### PR TITLE
chore(install): fetch license key from nerdgraph during profile initialization

### PR DESCRIPTION
**Pull request by ctrombley**

### CODE REVIEW

**Review by devfreddy — DISMISSED** [review]
🌮

File: internal/install/install.go [hunk 1]
```diff
@@ -82,6 +85,21 @@ func executeRecipeSteps(r recipe) error {
...
 	e.Taskfile.Vars.Merge(globals)
 
+	credentials.WithCredentials(func(c *credentials.Credentials) {
+		v := taskfile.Vars{}
+		licenseKey := c.Profiles[c.DefaultProfile].LicenseKey
+		if licenseKey == "" {
+			err = errors.New("license key not found in default profile")
+		}
+
+		v.Set("NR_LICENSE_KEY", taskfile.Var{Static: licenseKey})
```
Comment (zlesnr) [inline internal/install/install.go:95]: This is cool.

**Review by zlesnr — COMMENTED** [review]
This looks great.  A few places I might want to have a log message.

File: cmd/newrelic/command.go [hunk 2]
```diff
@@ -57,56 +49,57 @@ func initializeCLI(cmd *cobra.Command, args []string) {
...
-						accountID = resp[0].ID
-					} else {
-						log.Warnf("more than one account found")
-					}
-				})
+			if licenseKey == "" {
+				// We should have an account ID by now, so fetch the license key for it.
+				licenseKey, err = fetchLicenseKey(nrClient, accountID)
+				if err != nil {
+					return
```
[the remark this answers is not in the document]
Comment (jbeveland27) [reply cmd/newrelic/command.go:90]: Additional thought to this: if you set an error message [here](https://github.com/newrelic/newrelic-cli/pull/440/files?file-filters%5B%5D=.go#diff-62793eff3b471e494935d875be8b0e5b4907a288ea072e955e3685ab6b94e920R145), that would propagate back up and get displayed from the log message [here](https://github.com/newrelic/newrelic-cli/pull/440/files?file-filters%5B%5D=.go#diff-62793eff3b471e494935d875be8b0e5b4907a288ea072e955e3685ab6b94e920R122).

**Comment by jbeveland27** [comment]
Looks solid to me 🏆

File: cmd/newrelic/command.go [hunk 3]
```diff
@@ -57,56 +49,57 @@ func initializeCLI(cmd *cobra.Command, args []string) {
 			return
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply cmd/newrelic/command.go:49]: 👍

File: cmd/newrelic/command.go [hunk 4]
```diff
@@ -57,56 +49,57 @@ func initializeCLI(cmd *cobra.Command, args []string) {
...
-					return true
-				}
-			}
+		region = os.Getenv("NEW_RELIC_REGION")
+		licenseKey = os.Getenv("NEW_RELIC_LICENSE_KEY")
 
-			return false
+		// If we don't have a personal API key we can't initialize a profile.
+		if apiKey == "" {
+			return
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply cmd/newrelic/command.go:60]: 👍

File: cmd/newrelic/command.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ctrombley) [reply cmd/newrelic/command.go:90]: Leaving as-is per Justin's comment

**Review by zlesnr — APPROVED** [review]

**Pull request merged**